pkg/discovery/dtofactory/pod_entity_dto_builder.go

```diff
@@ -252,21 +252,27 @@ func (builder *podEntityDTOBuilder) getPodCommoditiesBought(pod *api.Pod, cpuFre
 }
 
 // Build the CommodityDTOs bought by the pod from the quota provider.
-func (builder *podEntityDTOBuilder) getPodCommoditiesBoughtFromQuota(pod *api.Pod, cpuFrequency float64) ([]*proto.CommodityDTO, error) {
+func (builder *podEntityDTOBuilder) getPodCommoditiesBoughtFromQuota(quotaUID string, pod *api.Pod, cpuFrequency float64) ([]*proto.CommodityDTO, error) {
 	var commoditiesBought []*proto.CommodityDTO
 	key := util.PodKeyFunc(pod)
 
-	// cpu and cpu provisioned needs to be converted from number of cores to frequency.
+	// cpu allocation needs to be converted from number of cores to frequency.
 	converter := NewConverter().Set(func(input float64) float64 {
 		return input * cpuFrequency
-	}, metrics.CPU, metrics.CPULimit)
+	}, metrics.CPULimit)
 
 	// Resource Commodities.
-	resourceCommoditiesBought, err := builder.getResourceCommoditiesBought(metrics.PodType, key, podResourceCommodityBoughtFromQuota, converter, nil)
-	if err != nil {
-		return nil, err
+	for _, resourceType := range podResourceCommodityBoughtFromQuota {
+		commBought, err := builder.getResourceCommodityBoughtWithKey(metrics.PodType, key,
+			resourceType, quotaUID, converter, nil)
+		if err != nil {
+			// skip this commodity
+			glog.Errorf("%s::%s: cannot build sold commodity %s : %s",
+				metrics.PodType, key, resourceType, err)
+			continue
```

Comment:
To prevent from unexpected happening, should abort PodEntity building on error, 
as the latest [PR](https://github.com/turbonomic/kubeturbo/pull/166) does.

Unexpected thing:
If neither VCPU nor VMEM commodity bought from VDC get built, then market will assume that this pod does not need to buy any commodity from VDC.